storage/innobase/dict/dict0stats.cc

```diff
@@ -1372,6 +1379,77 @@ static dberr_t btr_pcur_open_level(btr_pcur_t *pcur, ulint level, mtr_t *mtr,
 }
 
 
+/** Stores various statistical metrics gathered during an index scan
+or fetch operation. */
+struct IndexScanStatistics
+{
+  /* Array for number of distinct keys for all prefixes */
+  ib_uint64_t *n_diff_on_level = nullptr;
+
+  /* Total number of records */
+  ib_uint64_t total_recs = 0;
+
+  /* Total number of pages */
+  ib_uint64_t total_pages = 0;
+
+  /* Boundaries of the groups of distinct keys */
+  boundaries_t *n_diff_boundaries = nullptr;
+
+  /* Array of number of non-null keys for all prefixes */
+  ib_uint64_t *n_non_null_on_level = nullptr;
+
+  /* Whether nulls are considered unequal for statistics */
+  bool nulls_unequal = false;
+
+  IndexScanStatistics(bool nulls_unequal_): nulls_unequal(nulls_unequal_) {}
+  IndexScanStatistics(bool nulls_unequal_, bool non_null,
+                      dict_index_t *index)
+  : nulls_unequal(nulls_unequal_)
+  {
+    memset(index->stat_n_diff_key_vals, 0x0,
+	   index->n_uniq * sizeof(index->stat_n_diff_key_vals[0]));
+    n_diff_on_level= index->stat_n_diff_key_vals;
```

Comment:
Why are `n_diff_on_level` and `n_non_null_on_level` not initialized in the initializer list?